### PR TITLE
fix: duplicate eventmanager registration detection

### PR DESCRIPTION
**Pull request by xtrm-en**
## Description
Adds a guard clause to `EventManager#register` to prevent duplicate registration.

This was suggested on discord by `godofthegoths#5017`, please create an issue next time.

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

**Review by nxtdaydelivery — COMMENTED** [review]
why not just have a map here? its not like we are mixing someone else's event bus

**Comment by xtrm-en** [comment]
do you mean having a separate map/list that adds/remove listeners at the same time the EventBus does? 
idk i just did it like that lol.

**Comment by nxtdaydelivery** [comment]
but reflection slow 😭

**Comment by xtrm-en** [comment]
yeah but it's on initialization it shouldn't matter

**Comment by Wyvest** [comment]
> yeah but it's on initialization it shouldn't matter

please rewrite it to not use reflection lol, a lot of mods (including our own) call it on tick events and stuff and that can slow things down severely

**Review by Wyvest — DISMISSED** [review]

**Review by Wyvest — APPROVED** [review]

**Review by nxtdaydelivery — APPROVED** [review]
good looks to me

**Review by Wyvest — APPROVED** [review]

File: src/main/java/cc/polyfrost/oneconfig/events/EventManager.java [hunk 1]
```diff
@@ -40,11 +45,9 @@ public final class EventManager {
      * The instance of the {@link EventManager}.
      */
     public static final EventManager INSTANCE = new EventManager();
+    private static final Logger LOGGER = LogManager.getLogger("OneConfig/EventManager");
     private final EventBus eventBus = new EventBus(new LMFInvoker(), new OneConfigExceptionHandler());
-
-    private EventManager() {
```
Comment (nxtdaydelivery) [inline src/main/java/cc/polyfrost/oneconfig/events/EventManager.java:45]: why?

**Review by nxtdaydelivery — APPROVED** [review]
a

File: src/main/java/cc/polyfrost/oneconfig/events/EventManager.java (hunk 1 above)
Comment (xtrm-en) [reply src/main/java/cc/polyfrost/oneconfig/events/EventManager.java:45]: whoops i guess?

**Pull request merged**